modules/ocl/feature_match.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * cv_feature_match.h - optical flow feature match
+ *
+ *  Copyright (c) 2016-2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Wind Yuan <feng.yuan@intel.com>
+ * Author: Yinhang Liu <yinhangx.liu@intel.com>
+ */
+
+#ifndef XCAM_FEATURE_MATCH_H
+#define XCAM_FEATURE_MATCH_H
+
+#include <base/xcam_common.h>
+#include <base/xcam_buffer.h>
+#include <dma_video_buffer.h>
+#include <smartptr.h>
+#include "xcam_obj_debug.h"
+#include "image_file_handle.h"
+#include "ocl/cv_base_class.h"
+#include "interface/data_types.h"
+
+#include <ocl/cl_context.h>
+#include <ocl/cl_device.h>
+#include <ocl/cl_memory.h>
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/ocl.hpp>
```

Comment:
Move all opencv headers to cv_feature_match and cv_capi_feature_match.
and cv_base_class should be only used in cv_feature_match.
ocl/cl_xxx.h should be removed here.

Reply:
removed.